UnityProject/Assets/Scripts/UI/Cargo/GUI_SecurityRecords.cs

```diff
@@ -0,0 +1,162 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using UnityEngine;
+using UnityEngine.EventSystems;
+using UnityEngine.UI;
+
+public class GUI_SecurityRecords : NetTab
+{
+	[SerializeField]
+	private NetPageSwitcher nestedSwitcher;
+	[SerializeField]
+	private GUI_SecurityRecordsEntriesPage entriesPage;
+	[SerializeField]
+	private GUI_SecurityRecordsEntryPage entryPage;
+	[SerializeField]
+	private NetLabel idText;
+	private SecurityRecordsConsole console;
+	public IDCard InsertedCard;
+
+	public override void OnEnable()
+	{
+		base.OnEnable();
+		StartCoroutine(WaitForProvider());
+	}
+
+	IEnumerator WaitForProvider()
+	{
+		while (Provider == null)
+		{
+			yield return WaitFor.EndOfFrame;
+		}
+
+		if (CustomNetworkManager.Instance._isServer)
+		{
+			console = Provider.GetComponentInChildren<SecurityRecordsConsole>();
+			console.OnConsoleUpdate.AddListener(UpdateScreen);
+		}
+	}
+
+	public void UpdateScreen()
+	{
+		if (InsertedCard == null && console.IdCard != null)
+			InsertId(console.IdCard);
+
+		if (nestedSwitcher.CurrentPage == entriesPage)
+			entriesPage.IdNameUpdate();
+		else if (nestedSwitcher.CurrentPage == entryPage)
+			entryPage.IdNameUpdate();
+		else
+			UpdateIdText();
+	}
+
+	/// <summary>
+	/// Insert some ID into console and update login details.
+	/// Will spit out currently inserted ID card.
+	/// </summary>
+	///<param name="cardToInsert">Card you want to insert</param>
+	private void InsertId(IDCard cardToInsert)
+	{
+		if (InsertedCard != null)
+			RemoveId();
+		InsertedCard = cardToInsert;
+	}
+
+	/// <summary>
+	/// Spits out ID card from console and updates login details.
+	/// </summary>
+	public void RemoveId()
+	{
+		if (console != null && InsertedCard == null && console.IdCard != null)
+			InsertedCard = console.IdCard;
+		if (InsertedCard == null)
+			return;
```

Comment:
you already checked if InsertedCard is null, you can remove this
edit: actually, why is InsertedCard being set in RemoveId() ?